--- cantata-2.5.0.orig2/CMakeLists.txt	2021-10-06 19:18:08.000000000 +0200
+++ cantata-2.5.0.fix/CMakeLists.txt	2026-05-20 18:22:01.794069835 +0200
@@ -124,7 +124,7 @@
 endif ()
 
 if (NOT APPLE AND NOT WIN32)
-    if (CANTATA_HELPERS_LIB_DIR AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+    if (CANTATA_HELPERS_LIB_DIR)
         set(LINUX_LIB_DIR ${CANTATA_HELPERS_LIB_DIR})
     else ()
         set(LINUX_LIB_DIR lib)
