--- cantata-1.5.2/CMakeLists.txt	2016-01-11 07:35:12.310535875 +0100
+++ cantata-2.0.1/CMakeLists.txt	2016-07-12 13:06:24.291313896 +0300
@@ -146,9 +146,9 @@
 endif ()
 
 if (NOT APPLE AND NOT WIN32)
-    if (CANTATA_HELPERS_LIB_DIR AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+    if (CANTATA_HELPERS_LIB_DIR)
         set(LINUX_LIB_DIR ${CANTATA_HELPERS_LIB_DIR})
-    else ()
+    else ()
         set(LINUX_LIB_DIR lib)
     endif ()
 endif ()
