diff -urN cantata-2.5.0.orig/CMakeLists.txt cantata-2.5.0/CMakeLists.txt
--- cantata-2.5.0.orig/CMakeLists.txt	2021-10-06 19:18:08.000000000 +0200
+++ cantata-2.5.0/CMakeLists.txt	2026-05-20 18:21:40.659621108 +0200
@@ -540,10 +540,10 @@
 
 add_subdirectory(translations)
 add_subdirectory(support)
-add_subdirectory(3rdparty/qtiocompressor)
+
 add_subdirectory(streams/icons)
 add_subdirectory(online/icons)
-target_link_libraries(cantata support-core qtiocompressor ${CANTATA_LIBS} ${QTLIBS} ${ZLIB_LIBRARIES})
+target_link_libraries(cantata support-core Qt5Solutions_IOCompressor-2.3 ${CANTATA_LIBS} ${QTLIBS} ${ZLIB_LIBRARIES})
 
 # enable warnings
 add_definitions(-DQT_NO_DEBUG_OUTPUT)
diff -urN cantata-2.5.0.orig/context/albumview.cpp cantata-2.5.0/context/albumview.cpp
--- cantata-2.5.0.orig/context/albumview.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/context/albumview.cpp	2026-05-20 18:21:40.632708014 +0200
@@ -26,7 +26,7 @@
 #include "gui/covers.h"
 #include "network/networkaccessmanager.h"
 #include "support/utils.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "contextengine.h"
 #include "widgets/textbrowser.h"
 #include "widgets/icons.h"
diff -urN cantata-2.5.0.orig/context/artistview.cpp cantata-2.5.0/context/artistview.cpp
--- cantata-2.5.0.orig/context/artistview.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/context/artistview.cpp	2026-05-20 18:21:40.635414108 +0200
@@ -26,7 +26,7 @@
 #include "gui/apikeys.h"
 #include "support/utils.h"
 #include "network/networkaccessmanager.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "widgets/textbrowser.h"
 #include "widgets/icons.h"
 #include "contextengine.h"
diff -urN cantata-2.5.0.orig/context/songview.cpp cantata-2.5.0/context/songview.cpp
--- cantata-2.5.0.orig/context/songview.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/context/songview.cpp	2026-05-20 18:21:40.637695666 +0200
@@ -43,7 +43,7 @@
 #include "widgets/textbrowser.h"
 #include "gui/stdactions.h"
 #include "mpd-interface/mpdstatus.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include <QFile>
 #include <QDir>
 #include <QFileInfo>
diff -urN cantata-2.5.0.orig/context/wikipediasettings.cpp cantata-2.5.0/context/wikipediasettings.cpp
--- cantata-2.5.0.orig/context/wikipediasettings.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/context/wikipediasettings.cpp	2026-05-20 18:21:40.639973022 +0200
@@ -27,7 +27,7 @@
 #include "support/icon.h"
 #include "support/spinner.h"
 #include "gui/settings.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "support/utils.h"
 #include "support/action.h"
 #include "support/thread.h"
diff -urN cantata-2.5.0.orig/models/musiclibraryitemroot.cpp cantata-2.5.0/models/musiclibraryitemroot.cpp
--- cantata-2.5.0.orig/models/musiclibraryitemroot.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/models/musiclibraryitemroot.cpp	2026-05-20 18:21:40.642334271 +0200
@@ -31,7 +31,7 @@
 #include "musiclibrarymodel.h"
 #include "mpd-interface/mpdparseutils.h"
 #include "mpd-interface/mpdconnection.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include <QXmlStreamReader>
 #include <QXmlStreamWriter>
 #include <QFile>
diff -urN cantata-2.5.0.orig/models/streamsmodel.cpp cantata-2.5.0/models/streamsmodel.cpp
--- cantata-2.5.0.orig/models/streamsmodel.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/models/streamsmodel.cpp	2026-05-20 18:21:40.644692898 +0200
@@ -34,7 +34,7 @@
 #include "support/action.h"
 #include "gui/stdactions.h"
 #include "support/actioncollection.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "config.h"
 #include "support/globalstatic.h"
 #include <QModelIndex>
diff -urN cantata-2.5.0.orig/online/onlinedbservice.cpp cantata-2.5.0/online/onlinedbservice.cpp
--- cantata-2.5.0.orig/online/onlinedbservice.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/online/onlinedbservice.cpp	2026-05-20 18:21:40.647055668 +0200
@@ -25,7 +25,7 @@
 #include "gui/covers.h"
 #include "models/roles.h"
 #include "network/networkaccessmanager.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "db/onlinedb.h"
 #include <QXmlStreamReader>
 
diff -urN cantata-2.5.0.orig/online/podcastservice.cpp cantata-2.5.0/online/podcastservice.cpp
--- cantata-2.5.0.orig/online/podcastservice.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/online/podcastservice.cpp	2026-05-20 18:21:40.650320301 +0200
@@ -33,7 +33,7 @@
 #include "mpd-interface/mpdconnection.h"
 #include "config.h"
 #include "http/httpserver.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "network/networkaccessmanager.h"
 #include "models/roles.h"
 #include "models/playqueuemodel.h"
diff -urN cantata-2.5.0.orig/scrobbling/scrobbler.cpp cantata-2.5.0/scrobbling/scrobbler.cpp
--- cantata-2.5.0.orig/scrobbling/scrobbler.cpp	2022-01-08 22:24:03.000000000 +0100
+++ cantata-2.5.0/scrobbling/scrobbler.cpp	2026-05-20 18:21:40.652977349 +0200
@@ -34,7 +34,7 @@
 #include "support/globalstatic.h"
 #include "support/utils.h"
 #include "support/configuration.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include <QUrl>
 #include <QStringList>
 #include <QCryptographicHash>
